manager/testdata/new-blueprints/read-transform-blueprint.yaml

```diff
@@ -0,0 +1,92 @@
+apiVersion: app.m4d.ibm.com/v1alpha2
+kind: Blueprint
+metadata:
+  name: read-and-transform
+  labels:
+spec:
+  cluster: theshire
+  appSelector: "app:demoapp"
+  appCluster: theshire
+  steps:
+  - name: read-data
+    assetID: m4d-notebook-sample/paysim-csv
+    templateSpec:
+      chart: ghcr.io/mesh-for-data/m4d-arrow-flight-common-read-conf:0.1.0
+      m4dModule: arrow-flight-common-read-conf # This is the name of the deployed M4DModule definition
+    moduleLabel: "mod: common-arrow-flight-read-service"
+    source:
+      vault:
+        address: http://vault.m4d-system:8200
+        authPath: /v1/auth/kubernetes/login
+        role: module
+        secretPath: /v1/kubernetes-secrets/paysim-csv?namespace=m4d-notebook-sample
+      connection:
+        s3:
+          endpoint: localhost:8001
+          bucket: srcbucket
+          object: data.parq
+    service:
+      name: common-arrow-flight-read-service
+      assetID: m4d-notebook-sample/paysim-csv # Differentiator to original asset for multi user arrow flight service? How to best specify?
+      port: 8080
+      interfaceDetails:
+        protocol: arrow-flight
+        dataformat: arrow
+    transformations: [] # No transformations done in this module
+  - name: transform-step
+    assetID: m4d-notebook-sample/paysim-csv
+    templateSpec:
+      chart: ghcr.io/mesh-for-data/m4d-transform-conf-arrow-flight:0.1.0
+      moduleLabel: "mod: common-transform-arrow-flight"
+    source:
+      step: read-data # the name of step for debugging purposes
+      selector: "mod: common-arrow-flight-read-service" # Selector of the dependency
+      connection:
+        service:
```

Comment:
is this an indent issue for service? or what is service under "connection"?

Reply:
This is the source specification for the transform step. The source in this case is a service that is using arrow-flight protocol etc that is defined in the `read-data` step. I think it's easier to reuse existing fields explicitly here to state that there is a dependency to the first step.